src/dsync/dsync.c

```diff
@@ -2410,42 +2953,75 @@ int main(int argc, char **argv)
     }
     mfu_flist_walk_param_paths(1, destpath, walk_opts, flist2);
 
+    if (options.link_dest != NULL) {
+        mfu_flist_walk_param_paths(1, link_dest_path, walk_opts, flist_link_dest);
+    }
+
     /* store src and dest path strings */
     const char* path1 = srcpath->path;
     const char* path2 = destpath->path;
 
+    char* path3 = NULL;
+    if (options.link_dest != NULL) {
+        path3 = link_dest_path->path;
+    }
+
     /* map files to ranks based on portion following prefix directory */
     mfu_flist flist3 = mfu_flist_remap(flist1, (mfu_flist_map_fn)dsync_map_fn, (const void*)path1);
     mfu_flist flist4 = mfu_flist_remap(flist2, (mfu_flist_map_fn)dsync_map_fn, (const void*)path2);
 
+    mfu_flist flist5 = NULL;
+
+    if (options.link_dest != NULL) {
+        flist5 = mfu_flist_remap(flist_link_dest, (mfu_flist_map_fn)dsync_map_fn, (const void*)path3);
+    }
+
     /* free original file lists */
     mfu_flist_free(&flist1);
     mfu_flist_free(&flist2);
+    if (options.link_dest != NULL) {
+        mfu_flist_free(&flist_link_dest);
+    }
 
     /* map each file name to its index and its comparison state */
     strmap* map1 = dsync_strmap_creat(flist3, path1);
     strmap* map2 = dsync_strmap_creat(flist4, path2);
+    strmap* map3 = NULL;
 
     /* compare files in map1 with those in map2 */
```

Comment:
This comment got separated from its code block, so let's move that back down to the strmap_compare() block.

Reply:
Done